### PR TITLE
[JENKINS-64107] Convert SSHD module to a plugin

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-55582](https://issues.jenkins-ci.org/browse/JENKINS-55582) Convert modules to plugins
[JENKINS-64107](https://issues.jenkins-ci.org/browse/JENKINS-64107) Convert SSHD module to a plugin
[JEP](https://github.com/jenkinsci/jep/pull/326) Convert modules to plugins

* This PR change to use the  sshd-plugin
* Bump the Core to the 2.289.1

related to https://github.com/jenkinsci/sshd-plugin/pull/46

### CODE REVIEW

**Comment by kuisathaverat** [comment]
waiting for the core 2.282 to be released

File: pom.xml
Comment (kuisathaverat) [inline pom.xml:64]:
```suggestion
      <version>3.0.3</version>
```

**Comment by kuisathaverat** [comment]
@jglick Do you have any idea what this can happen on the CI? locally it works ok 

```
[2021-03-13T14:19:11.718Z] [ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:validate-hpi (default-validate-hpi) on project git-server: Dependency org.jenkins-ci.modules:sshd:jar:3.0.3 requires Jenkins 2.282 or higher. -> [Help 1]

[2021-03-13T14:19:11.718Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:validate-hpi (default-validate-hpi) on project git-server: Dependency org.jenkins-ci.modules:sshd:jar:3.0.3 requires Jenkins 2.282 or higher.
```

**Comment by kuisathaverat** [comment]
@jglick Do you have any idea what this can happen on the CI? locally it works ok 

```
[2021-03-13T14:19:11.718Z] [ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:validate-hpi (default-validate-hpi) on project git-server: Dependency org.jenkins-ci.modules:sshd:jar:3.0.3 requires Jenkins 2.282 or higher. -> [Help 1]

[2021-03-13T14:19:11.718Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:validate-hpi (default-validate-hpi) on project git-server: Dependency org.jenkins-ci.modules:sshd:jar:3.0.3 requires Jenkins 2.282 or higher.
```

**Comment by jglick** [comment]
If you lack write access, your changes to `Jenkinsfile` will be ignored.

**Comment by jglick** [comment]
merge with `master` after #28

File: Jenkinsfile
Comment (timja) [inline Jenkinsfile:1]: Just revert changes to this file?

Comment (kuisathaverat) [reply Jenkinsfile]: done

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
@jglick ^^

File: pom.xml
Comment (jglick) [inline pom.xml:46]:
```suggestion
        <artifactId>bom-2.277.x</artifactId>
```

?

**Review by jglick — APPROVED** [review]
Possible mistake in BOM selection; otherwise looks good. Seems I have permission to release, too.

**Comment by kuisathaverat** [comment]
require to merge an release https://github.com/jenkinsci/sshd-plugin/pull/37

File: pom.xml
Comment (kuisathaverat) [inline pom.xml:64]: Change to the released version after merge https://github.com/jenkinsci/sshd-plugin/pull/37

```suggestion
      <version>3.0.4</version>
```

Comment (jglick) [reply pom.xml:64]: So this should be draft status until then.

**Review by jglick — APPROVED** [review]

**Review by rsandell — APPROVED** [review]

File: pom.xml
Comment (kuisathaverat) [inline pom.xml:64]:
```suggestion
      <version>3.0.4-rc203.a16405ea4f48</version>
```

**Comment by kuisathaverat** [comment]
It passes the CI and I have manually tested it.

File: pom.xml
Comment (kuisathaverat) [inline pom.xml:23]: Use latest LTS
```suggestion
    <jenkins.version>2.289.1</jenkins.version>
```

**Comment by kuisathaverat** [comment]
@alecharp  @varyvol @batmat @fcojfernandez @MRamonLeon @rsandell @jglick I summon you all as maintainers of the plugin to coordinate the merge and release of this change. This is a breaking change after I will merge and release https://github.com/jenkinsci/sshd-plugin/pull/46 so we have to make the releases of both plugins at the same time to avoid issues. How and when it is posible to make the merge and release?

**Comment by jglick** [comment]
According to RPU no one permission has release permission on both I am afraid. I do have release permission here, so I am willing to cut a release of this plugin as soon as the upstream is released. (I am in US-EDT TZ.)

I assume you have done the end-to-end sanity testing. (Using an actual Jenkins installation, not just `hpi:run`.)

**Comment by kuisathaverat** [comment]
>I assume you have done the end-to-end sanity testing. (Using an actual Jenkins installation, not just hpi:run.)

of course, I have made some manual testing with a real Jenkins instance.

>According to RPU no one permission has release permission on both I am afraid. I do have release permission here, so I am willing to cut a release of this plugin as soon as the upstream is released. (I am in US-East TZ.)

I am in CEST TZ so right now we have 6 hours of difference, I can release the sshd plugin this week any day from 15:00 to 19:00 CEST TZ (09:00-13:00 EDT TZ), just tell me what day and time is best for you.

**Comment by jglick** [comment]
Yesterday was a US holiday but I am available for work today. Shall we say in about 90m ~ 2021-07-06T15:30+00:00?

**Comment by jglick** [comment]
@kuisathaverat let me know. I should be watching GitHub notifications.

**Comment by kuisathaverat** [comment]
@jglick I saw it too late. I will ping you tomorrow at your 08:00 AM and let's see if we can find a gap to release both plugins

**Comment by jglick** [comment]
Fine, I should be available pretty much all day.

**Comment by kuisathaverat** [comment]
@jglick Are you ready? if so I will merge and release https://github.com/jenkinsci/sshd-plugin/pull/46

**Comment by jglick** [comment]
@kuisathaverat yes, go for it

File: pom.xml
Comment (jglick) [inline pom.xml:64]: to be updated

Comment (kuisathaverat) [reply pom.xml:64]:
```suggestion
      <version>3.0.4</version>
```

File: pom.xml
Comment (jglick) [inline pom.xml:64]: BTW the SemVer-like version scheme is totally misleading here. Maybe better to just give up and switch to JEP-229.

**Comment by kuisathaverat** [comment]
@jglick I have released sshd plugin and updated the version here to 3.0.4, the CI has passed.

File: pom.xml
Comment (kuisathaverat) [reply pom.xml:64]: sure, I want to change all my plugins to JEP-229.

**Comment by jglick** [comment]
https://github.com/jenkinsci/git-server-plugin/releases/tag/git-server-1.10

**Pull request merged**
